hardhat.config.js

```diff
@@ -19,218 +19,265 @@ module.exports = {
+        process.env.DEPLOYER_ADDRESS, // specify Ledger Address here (required) ADMIN ADDRESS
+      ],
       timeout: 600000,
     },
     optimism: {
       chainId: 10,
-      url: "http://127.0.0.1:1248", // this is the RPC endpoint exposed by Frame
+      url: "https://mainnet.optimism.io", 
+      ledgerAccounts: [
+        process.env.DEPLOYER_ADDRESS, // specify Ledger Address here (required) ADMIN ADDRESS
+      ],
       timeout: 600000,
     },
     arbitrumOne: {
         chainId: 42161,
-        url: "http://127.0.0.1:1248", // this is the RPC endpoint exposed by Frame
+        url: "https://arb1.arbitrum.io/rpc", 
+        ledgerAccounts: [
+        process.env.DEPLOYER_ADDRESS, // specify Ledger Address here (required) ADMIN ADDRESS
+      ],
         timeout: 600000,
       },
       base: {
         chainId: 8453,
-        url: "http://127.0.0.1:1248", // this is the RPC endpoint exposed by Frame
+        url: "https://mainnet.base.org", 
+        ledgerAccounts: [
+        process.env.DEPLOYER_ADDRESS, // specify Ledger Address here (required) ADMIN ADDRESS
+      ],
         timeout: 600000,
       },
       linea: {
         chainId: 59144,
-        url: "http://127.0.0.1:1248", // this is the RPC endpoint exposed by Frame
+        url: "https://rpc.linea.build", 
+        ledgerAccounts: [
+        process.env.DEPLOYER_ADDRESS, // specify Ledger Address here (required) ADMIN ADDRESS
+      ],
         timeout: 600000,
       },
       scroll: {
         chainId: 534352,
-        url: "http://127.0.0.1:1248", // this is the RPC endpoint exposed by Frame
+        url: "https://rpc.scroll.io/", 
+        ledgerAccounts: [
+        process.env.DEPLOYER_ADDRESS, // specify Ledger Address here (required) ADMIN ADDRESS
+      ],
         timeout: 600000,
       },
     //Testnets
     europaTestnet: {
         chainId: 476158412,
         url: "https://staging-v3.skalenodes.com/v1/staging-legal-crazy-castor",
-        accounts: [process.env.DEPLOYER_KEY || "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"],
+        accounts: [process.env.TESTNET_DEPLOYER_KEY || "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"],
         timeout: 600000,
       },
       calypsoTestnet: {
         chainId: 344106930,
         url: "https://staging-v3.skalenodes.com/v1/staging-utter-unripe-menkar",
-        accounts: [process.env.DEPLOYER_KEY || "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"],
+        accounts: [process.env.TESTNET_DEPLOYER_KEY || "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"],
         timeout: 600000,
       },
     xdcTestnet: {
       chainId: 51,
       url: "https://erpc.apothem.network",
-      accounts: [process.env.DEPLOYER_KEY || "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"],
+      accounts: [process.env.TESTNET_DEPLOYER_KEY || "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"],
```

Comment:
|| "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"

this part seems unnecessary. you can create a .env.sample file and prepopulate TESTNET_DEPLOYER_KEY .

bad idea to hardcode private keys in codebase.

Reply:
this is just the default hardhat local deployer key, sure can make this change, we don't need it can keep it as an empty string.